packages/dx-react-grid-demos/src/demo-sources/optimizing-performance/bootstrap4/chart-connect-props.jsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import Paper from '@material-ui/core/Paper';
+import { connectProps } from '@devexpress/dx-react-core';
+import {
+  Scale, EventTracker, SelectionState,
+} from '@devexpress/dx-react-chart';
+import {
+  Chart,
+  SplineSeries,
+  ArgumentAxis,
+  ValueAxis,
+  Legend,
+} from '@devexpress/dx-react-chart-bootstrap4';
+import { energyConsumption as data } from '../../../demo-data/chart-data';
+
+const LegendRoot = ({ classes, ...restProps }) => (
+  <Legend.Root
+    {...restProps}
+    style={{
```

Comment:
On each *LegendRoot* render new (while actually the same) style property value is created.
Though in general it would be tolerable because component itself is rather simple, in a *performance optimization* 
demo it should be avoided.

```javascript
const legendRootStyle = {
  display: 'flex',
  margin: 'auto',
  flexDirection: 'row'
};
const LegendRoot = props => (
  <Legend.Root {...props} style={legendRootStyle} />
);
```